### PR TITLE
refactor operationManager with appModel

### PR DESCRIPTION
**Pull request by tanbowensg**
I replace all the `ComponentSchema[]` to `AppModel` in every operations so that operation don't have convert  `ComponentSchema[]` to `AppModel` every time.

### CODE REVIEW

File: packages/editor/src/operations/branch/createComponentBranchOperation.ts [hunk 1]
```diff
@@ -17,10 +17,10 @@ export type CreateComponentBranchOperationContext = {
 };
 
 export class CreateComponentBranchOperation extends BaseBranchOperation<CreateComponentBranchOperationContext> {
-  do(prev: ComponentSchema[]): ComponentSchema[] {
+  do(prev: AppModel): AppModel {
     // gen component id
     if (!this.context.componentId) {
-      this.context.componentId = genId(this.context.componentType, prev);
+      this.context.componentId = prev.genId(this.context.componentType as ComponentType);
```
Comment (Sczlog) [inline packages/editor/src/operations/branch/createComponentBranchOperation.ts:23]: should we narrow the type of context.componentType to ComponentType from string?

**Comment by Sczlog** [comment]
LGTM

File: packages/editor/src/AppModel/ComponentModel.ts [hunk 2]
```diff
@@ -170,9 +170,12 @@ export class ComponentModel implements IComponentModel {
   removeChild(child: IComponentModel) {
     const slotChildren = this.children[child.parentSlot!];
     if (slotChildren) {
-      slotChildren.splice(slotChildren.indexOf(child), 1);
-      child._isDirty = true;
-      this._isDirty = true;
+      const index = slotChildren.indexOf(child)
+      if (index > -1) {
```
Comment (Yuyz0112) [inline packages/editor/src/AppModel/ComponentModel.ts:174]: when index will be -1?

Comment (tanbowensg) [reply packages/editor/src/AppModel/ComponentModel.ts:174]: when `child` is not child of this component just having the same slotName.

**Review by Yuyz0112 — APPROVED** [review]

**Pull request merged**